frontend/src/marketplace/EntryList.tsx

```diff
@@ -22,49 +23,69 @@ export default function EntryList({
 }: EntryListProps) {
   const theme = useTheme()
 
-  const entriesDisplay = useMemo(() => {
-    return entries.map((entry, index) => (
-      <Fragment key={entry.id}>
-        <Stack direction='row'>
-          <Link
-            sx={{ textDecoration: 'none', whiteSpace: 'nowrap', textOverflow: 'ellipsis', overflow: 'hidden' }}
-            href={`${entry.kind}/${entry.id}`}
-          >
-            <Typography
-              variant='h5'
-              sx={{ fontWeight: '500', textDecoration: 'none', color: theme.palette.primary.main }}
+  const Row = memo(({ data, index, style }: { data: EntrySearchResult[]; index: number; style: CSSProperties }) => {
+    const entry = data[index]
+
+    return (
+      <Box
+        justifyContent='flex-start'
+        alignItems='center'
+        sx={{ backgroundColor: index % 2 ? theme.palette.container.main : 'none', p: 2, margin: 'auto', ...style }}
+        key={entry.id}
+      >
+        <Stack>
+          <Stack direction='row'>
+            <Link
+              sx={{ textDecoration: 'none', whiteSpace: 'nowrap', textOverflow: 'ellipsis', overflow: 'hidden' }}
+              href={`${entry.kind}/${entry.id}`}
             >
-              {entry.name}
-            </Typography>
-          </Link>
+              <Typography
+                variant='h5'
+                sx={{ fontWeight: '500', textDecoration: 'none', color: theme.palette.primary.main }}
+              >
+                {entry.name}
+              </Typography>
+            </Link>
+          </Stack>
+          <Typography
+            variant='body1'
+            sx={{ marginBottom: 2, whiteSpace: 'nowrap', textOverflow: 'ellipsis', overflow: 'hidden' }}
+          >
+            {entry.description}
+          </Typography>
+          <div>
```

Comment:
Is this `div` needed?

Reply:
It stops the chips in the `<ChipSelector />` component from also stacking incorrectly, however I feel we should probably add a stack inside `<ChipSelector />` to stop this from happening.

Reply:
Nevermind just realised that a `<Stack />` inside the chip component won't work either!